[[ Bug 17216 ]] Reinstate TIFF processing in Mac clipboard

runrevmark opened the pull request:
Some applications on Mac only present TIFF as the image format on
the clipboard (Preview included).

As the engine (as a whole) does not understand TIFF data, this
patch makes TIFF data on the clipboard masquerade as PNG data
allowing it to be used in a loss-less way (relative to the original
TIFF encoding).

The TIFF data will still be listed in the rawClipboardData, but
in the fullClipboardData it will come through as image and png.

runrevmark committed:
[[ Bug 17216 ]] Reinstate TIFF processing in Mac clipboard

Some applications on Mac only present TIFF as the image format on
the clipboard (Preview included).

As the engine (as a whole) does not understand TIFF data, this
patch makes TIFF data on the clipboard masquerade as PNG data
allowing it to be used in a loss-less way (relative to the original
TIFF encoding).

The TIFF data will still be listed in the rawClipboardData, but
in the fullClipboardData it will come through as image and png.

livecodefraser commented:
You ruined my beautiful code :cry: 

@livecode-vulcan review okay b3ded13

peter-b commented:
Clearly what is required here is an `class MCMacClipboard : public MCClipboard` that is used on Mac. :wink:

livecode-vulcan commented:
💙 review by @livecodefraser ok b3ded136a1a08fb9c67cf7f5c7740a192e9783ab

livecode-vulcan commented:
😞 test failure b3ded136a1a08fb9c67cf7f5c7740a192e9783ab
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2276)
- try-community-mac: [failure](https://vulcan.livecode.com/builders/try-community-mac/builds/2285)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2269)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2286)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2292)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2297)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2308)

runrevmark committed:
[[ Bug 17216 ]] Fix compilation on server.

The server engine doesn't have the TIFF processing code, so shouldn't
try to use it.

livecodefraser commented:
@livecode-vulcan review okay d366da4

livecode-vulcan commented:
💙 review by @livecodefraser ok d366da4fe2efa38f405253bee590c80a83773820

livecode-vulcan commented:
😎 test success d366da4fe2efa38f405253bee590c80a83773820
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2277)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2286)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2270)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2287)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2293)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2298)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2309)
